Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -1687,7 +1687,10 @@ void WebPageProxy::initializeWebPage(const Site& site, WebCore::SandboxFlags eff
     Ref browsingContextGroup = m_browsingContextGroup;
     Ref preferences = m_preferences;
 
-    m_mainFrame = WebFrameProxy::create(*this, browsingContextGroup->ensureProcessForSite(site, process, preferences), FrameIdentifier::generate(), effectiveSandboxFlags, ScrollbarMode::Auto,  openerFrame(protectedConfiguration()).get(), IsMainFrame::Yes);
+    Ref mainFrameProcess = browsingContextGroup->ensureProcessForSite(site, process, preferences);
```

Comment:
Rather than separating ensureProcessForSite from addFrameProcess, could we make ensureProcessForSite effectively call addFrameProcess and make a new function for provisional navigations?